### PR TITLE
Update imdb_top.py

### PR DESCRIPTION
**Pull request by naman1303**
Added validation on user input for zodiac sign and day. Also used tuple and made minor changes.

### CODE REVIEW

File: web_programming/daily_horoscope.py [hunk 1]
```diff
@@ -28,8 +28,18 @@ def horoscope(zodiac_sign: int, day: str) -> str:
         "11. Aquarius\n",
         "12. Pisces\n",
     )
-    zodiac_sign = int(input("number> ").strip())
-    print("choose some day:\n", "yesterday\n", "today\n", "tomorrow\n")
-    day = input("enter the day> ")
-    horoscope_text = horoscope(zodiac_sign, day)
-    print(horoscope_text)
+    zodiac_sign = int(input("Select a sign [1-12] > ").strip())
+    if zodiac_sign not in range(0, 13):
```
Comment (cclauss) [inline web_programming/daily_horoscope.py:32]: 0 is in range(0, 13)

File: web_programming/daily_horoscope.py [hunk 2]
```diff
@@ -28,8 +28,18 @@ def horoscope(zodiac_sign: int, day: str) -> str:
...
+        print("Invalid choice of Zodian sign!")
+    else:
+        days_list = {"1": "Yesterday", "2": "Today", "3": "Tomorrow"}
+        print("\nDays available:")
+        for day in days_list.items():
+            print("{}. {}".format(day[0], day[1]))
```
Comment (cclauss) [inline web_programming/daily_horoscope.py:38]:
```suggestion
            print("{}. {}".format(*day))
```

File: web_programming/daily_horoscope.py [hunk 3]
```diff
@@ -28,8 +28,18 @@ def horoscope(zodiac_sign: int, day: str) -> str:
...
+        inp_day = input("Choose a day [1-3] > ")
+        if inp_day not in days_list:
+            print("Invalid choice of day!")
+        else:
+            day = days_list[inp_day].lower()
```
Comment (cclauss) [inline web_programming/daily_horoscope.py:43]: Please condense these last 3 lines down to one line.

**Review by cclauss — APPROVED** [review]

File: web_programming/daily_horoscope.py (hunk 1 above)
Comment (naman1303) [reply web_programming/daily_horoscope.py]: Corrected it.

File: web_programming/daily_horoscope.py (hunk 2 above)
Comment (naman1303) [reply web_programming/daily_horoscope.py]: Thanks for this. Corrected.

File: web_programming/daily_horoscope.py (hunk 3 above)
Comment (naman1303) [reply web_programming/daily_horoscope.py]: I have made it one liner, still had to add an if condition to make horoscope function call.

File: web_programming/get_imdbtop.py [hunk 4]
```diff
@@ -8,8 +8,8 @@ def imdb_top(imdb_top_n):
         f"feature&sort=num_votes,desc&count={imdb_top_n}"
     )
     source = BeautifulSoup(requests.get(base_url).content, "html.parser")
-    for m in source.findAll("div", class_="lister-item mode-advanced"):
-        print("\n" + m.h3.a.text)  # movie's name
+    for i, m in enumerate(source.findAll("div", class_="lister-item mode-advanced")):
+        print("\n" + str(i+1) + ". " + m.h3.a.text)  # serial no and movie's name
```
Comment (cclauss) [inline web_programming/get_imdbtop.py]: Please use an f-string like line 15 does.

**Review by cclauss — CHANGES_REQUESTED** [review]
1. Please rename the filename to `imdb_top.py` to match the name of the function.
2. Let's have line 20 convert user input to an int.
3. Please add type hints to the function showing that `imdb_top_n` is an int and the function returns None.
4. Let's have the function raise a TypeError if `imdb_top_n` is not an int.
4. Please add a doctest that shows how `imdb_top()` behaves if `imdb_top_n` is `"I don't know."`.

**Review by jessica02200 — APPROVED** [review]

**Comment by naman1303** [comment]
> 1. Please rename the filename to `imdb_top.py` to match the name of the function.
> 2. Let's have line 20 convert user input to an int.
> 3. Please add type hints to the function showing that `imdb_top_n` is an int and the function returns None.
> 4. Let's have the function raise a TypeError if `imdb_top_n` is not an int.
> 5. Please add a doctest that shows how `imdb_top()` behaves if `imdb_top_n` is `"I don't know."`.

Except for 5th point, I have added all the changes. 
Read somewhere that for functions having user input, doctest is not a good idea. However, if I a good solution around this, will add it later.

File: web_programming/imdb_top.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            raise (e)
+
+
+def imdb_top(imdb_top_n):
+    """
+    Args:
+        imdb_top_n: Movie count provided by user. Integer value.
+    Returns:
+        None
+    """
```
Comment (cclauss) [inline web_programming/imdb_top.py:23]: Please look at other contributions in this repo to see how they do [type hints](https://docs.python.org/3/library/typing.html) and [doctests](https://docs.python.org/3/library/doctest.html).

Comment (naman1303) [reply web_programming/imdb_top.py]: Please check now.

**Review by 53jk1 — APPROVED** [review]

**Comment by naman1303** [comment]
@53jk1 - I have incorporated the requested changes by @cclauss already. However, the merging is blocked citing requested changes are to be addressed. I don't see any button to mark as resolved etc.
Please guide.

**Pull request closed without merging**